src/System.IO.MemoryMappedFiles/src/System/IO/MemoryMappedFiles/Interop.Windows.cs

```diff
@@ -0,0 +1,81 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using Microsoft.Win32.SafeHandles;
+using System;
+using System.IO;
+using System.Runtime.InteropServices;
+
+internal partial class Interop
+{
+    public static unsafe void CheckForAvailableVirtualMemory(ulong nativeSize)
+    {
+        Interop.mincore.MEMORYSTATUSEX memStatus;
+        memStatus.dwLength = (uint)sizeof(Interop.mincore.MEMORYSTATUSEX);
+        Interop.mincore.GlobalMemoryStatusEx(out memStatus);
+        ulong totalVirtual = memStatus.ullTotalVirtual;
+        if (nativeSize >= totalVirtual)
+        {
+            throw new IOException(SR.IO_NotEnoughMemory);
+        }
+    }
+
+    public static SafeMemoryMappedFileHandle CreateFileMapping(
+            SafeFileHandle hFile,
+            ref mincore.SECURITY_ATTRIBUTES securityAttributes,
+            int pageProtection,
+            long maximumSize,
+            string name)
+    {
+        // split the long into two ints
+        int capacityLow = unchecked((int)(maximumSize & 0x00000000FFFFFFFFL));
+        int capacityHigh = unchecked((int)(maximumSize >> 32));
```

Comment:
Sure.  I looked for a common utility helper and couldn't find one.  For now just put it in this file.